vald/evm/rpc/ethereum.go

```diff
@@ -68,6 +74,29 @@ func (c *EthereumClient) LatestFinalizedBlockNumber(ctx context.Context, confirm
 	return sdk.NewIntFromUint64(blockNumber).SubRaw(int64(confirmations)).AddRaw(1).BigInt(), nil
 }
 
+func (c *EthereumClient) TransactionReceipts(ctx context.Context, txHashes []common.Hash) ([]ReceiptResult, error) {
+	batch := slices.Map(txHashes, func(txHash common.Hash) rpc.BatchElem {
+		var receipt types.Receipt
+		return rpc.BatchElem{
+			Method: "eth_getTransactionReceipt",
+			Args:   []interface{}{txHash},
+			Result: &receipt,
+		}
+	})
```

Comment:
Do we want to check for duplicate tx hashes?

Reply:
I think the batch should query whatever get passed in, it's upon the event emitter (the network) to prevent duplicate tx hashes